core/test/element_arithmetic_test.cpp

```diff
@@ -243,3 +244,21 @@ TEST_F(ElementNanArithmeticTest, add_equals_with_rhs_int_lhs_int) {
   nan_add_equals(lhs, 2);
   EXPECT_EQ(3, lhs);
 }
+
+class ElementMatrixMatrixArithmeticTest : public ::testing::Test {
+protected:
+  Eigen::Matrix3d x = Eigen::Matrix3d::Identity();
+  Eigen::Matrix3d y = Eigen::Matrix3d::Identity();
```

Comment:
Identity matrix is not really a good test value, since test would pass even if multiplication is very wrong?